pkg/services/provisioning/alert_notifications/types.go

```diff
@@ -0,0 +1,116 @@
+package alert_notifications
+
+import (
+	"github.com/grafana/grafana/pkg/models"
+)
+import "github.com/grafana/grafana/pkg/components/simplejson"
+
+type configVersion struct {
+	ApiVersion int64 `json:"apiVersion" yaml:"apiVersion"`
+}
+
+type notificationsAsConfig struct {
+	ApiVersion int64
+
+	Notifications       []*notificationFromConfig
+	DeleteNotifications []*deleteNotificationConfig
+}
+
+type deleteNotificationConfig struct {
+	Name  string
+	OrgId int64
+}
+
+type notificationFromConfig struct {
+	OrgId int64
+
+	Name      string
+	Type      string
+	IsDefault bool
+	Settings  map[string]interface{}
+}
+
+type notificationsAsConfigV0 struct {
+	configVersion
+
+	Notifications       []*notificationFromConfigV0   `json:"alert_notifications" yaml:"alert_notifications"`
+	DeleteNotifications []*deleteNotificationConfigV0 `json:"delete_alert_notifications" yaml:"delete_alert_notifications"`
+}
+
+type deleteNotificationConfigV0 struct {
+	Name  string `json:"name" yaml:"name"`
+	OrgId int64  `json:"org_id" yaml:"org_id"`
+}
+
+type notificationFromConfigV0 struct {
```

Comment:
When we built provisioning for dashboards and datasources we didn't consider versions of the config files. So we had to retrofit in support for that. For alert notifiers, there is no point having multiple config structs. Yet :)

Reply:
Yeah, I agree, wanted to stay consistent with the rest. Would you rather that I remove the config version completely right now? This can always be brought back if needed in the future.